src/connectors/zenodo/zenodo_dataset_connector.py

```
@@ -0,0 +1,99 @@
+            creator = ", ".join(creators_list)  # TODO change field to an array
+        elif isinstance(record["creators"]["creator"]["creatorName"], str):
+            creator = record["creators"]["creator"]["creatorName"]
+
+        title = ""
+        if isinstance(record["titles"]["title"], str):
+            title = record["titles"]["title"]
+
+        description = ""
+        if isinstance(record["descriptions"]["description"], list):
+            for element in record["descriptions"]["description"]:
+                if element.get("@descriptionType") == "Abstract":
+                    description = element.get("#text")
+                    break
+        elif record["descriptions"]["description"]["@descriptionType"] == "Abstract":
+            description = record["descriptions"]["description"]["#text"]
+
+        date_published = None
+        date_format = "%Y-%m-%d"
+        if isinstance(record["dates"]["date"], list):
+            for element in record["dates"]["date"]:
+                if element.get("@dateType") == "Issued":
+                    date_string = element["#text"]
+                    date_published = datetime.strptime(date_string, date_format)
+                    break
+        elif record["dates"]["date"]["@dateType"] == "Issued":
+            date_string = record["dates"]["date"]["#text"]
+            date_published = datetime.strptime(date_string, date_format)
+
+        publisher = ""
+        if isinstance(record["publisher"], str):
+            publisher = record["publisher"]
+
+        # Get dataset keywords
+        keywords = []
+
+        if "subjects" in record:
+            if isinstance(record["subjects"]["subject"], str):
+                keywords = [record["subjects"]["subject"]]
+            elif isinstance(record["subjects"]["subject"], list):
+                keywords = [item for item in record["subjects"]["subject"] if isinstance(item, str)]
```

You're ignoring the keywords that are not strings. Why? What can they be? 
If you do not expect them to exist, it would be better to raise an error. Now we might miss data.

---

There is no a schema defined, the correct ones are dict and str, it is only for avoiding errors but if you prefer i can delete the conditional